FitGoal/FitGoal/UserProfileConfiguration/WalkthroughIconView.swift

```diff
@@ -41,6 +35,16 @@ class WalkthroughIconView: UIView {
         return background
     }()
     
+    var isSelected = false {
+        didSet {
+            if isSelected {
+                self.selectIconView()
+            } else {
+                self.deselectIconView()
+            }
```

Comment:
Instead of having this logic and two extra function, I think it would be clearer something like:

```swift
icon.selected = isSelected
backgroundView.isHidden = !isSelected
configureIcon(forState: icon.selected, withName: icon.name)
```

And then you can delete the selection functions.